src/tools/ceph_objectstore_tool.cc

```diff
@@ -974,6 +1149,49 @@ int ObjectStoreTool::get_object(ObjectStore *store, coll_t coll,
   return 0;
 }
 
+int dump_pg_metadata(Formatter *formatter, bufferlist &bl, metadata_section &ms)
+{
+  bufferlist::iterator ebliter = bl.begin();
+  ms.decode(ebliter);
+
+  formatter->open_object_section("metadata_section");
+
+  formatter->dump_unsigned("pg_disk_version", (int)ms.struct_ver);
+  formatter->dump_unsigned("map_epoch", ms.map_epoch);
+
+  formatter->open_object_section("OSDMap");
+  ms.osdmap.dump(formatter);
+  formatter->close_section();
+  formatter->flush(cout);
+  cout << std::endl;
+
+  formatter->open_object_section("info");
+  ms.info.dump(formatter);
+  formatter->close_section();
+  formatter->flush(cout);
+
+  formatter->open_object_section("log");
+  ms.log.dump(formatter);
+  formatter->close_section();
+  formatter->flush(cout);
+
+  formatter->open_object_section("pg_missing_t");
+  ms.missing.dump(formatter);
```

Comment:
@dzafman The compilation failure happens here - `metadata_section` does not have a member `missing`. If you immediately see how to fix, please advise?

Reply:
@dzafman If I'm parsing the git history correctly, it's because jewel does not have 44cb6d30640954c66fc03d4ce58b583ffd614951 and that commit looks pretty complicated to be backporting to jewel. So I will just remove this little section of code that deals with the missing set.